src/message.py

```diff
@@ -67,6 +67,17 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return str(self)
 
+    def __eq__(self, other) -> bool:
```

Comment:
not actionable in this PR: hmm, maybe we should consider making Message a dataclass.
This `__eq__` definition is fine as-is.

But in the long-term, it would be nice to switch to an automatically generated one so we don't have bugs if fields change, etc.

https://docs.python.org/3/library/dataclasses.html

Reply:
Left it as is for now. Might change it to a dataclass in a separate small PR.